server/src/main/java/io/spine/server/delivery/DeliveryMonitor.java

```diff
@@ -87,12 +87,52 @@ public boolean shouldContinueAfter(DeliveryStage stage) {
      */
     @SuppressWarnings("unused")  // This SPI method is designed for descendants.
     public void onDeliveryCompleted(DeliveryStats stats) {
-        // do nothing.
+        // Do nothing.
     }
 
     @SuppressWarnings({"WeakerAccess", "unused"})   /* Part of public API. */
     public void onDeliveryStarted(ShardIndex index) {
-        // do nothing.
+        // Do nothing.
+    }
+
+    /**
+     * Called if an {@code Exception} occurred when the {@code Delivery} attempted
+     * to pick up a shard.
+     *
+     * <p>Please note, this callback is <em>not</em> invoked in case the shard cannot be picked
+     * if it's already picked by another worker. It is so, because such a use case is a part
+     * of normal flow, and thus does not provoke a {@code RuntimeException}.
+     *
+     * <p>Returns an action to take in relation to the failure.
+     *
+     * <p>By default this callback returns an Action that does nothing. This means the occurred
+     * exception will be throw out of the {@code deliverMessagesFrom()} method.
+     *
+     * @param failure
+     *         contains an information about the occurred failure and gives access to
+     *         predefined {@code Action}s to handle the error.
+     */
+    @SuppressWarnings({"WeakerAccess", "unused"})  /* Part of public API. */
+    public FailedPickUp.Action onShardPickUpFailure(TechFailure failure) {
+        return failure.doNothing();
```

Comment:
I don't really like the idea to have `.doNothing()` meaning two different things. In this scenario, returning this value means "rethrow (or propagate) the exception", and in the scenario below it really means "do nothing".

Maybe we could have a more specific action-method for this exception-related scenario?